File: backend/app/schemas/label.py
"""Schemas Pydantic de etiquetas de Gmail vigiladas (spec §8.4)."""
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class GmailLabelAvailable(BaseModel):
    """Etiqueta real de Gmail traída por sync (aún no necesariamente vigilada)."""
    gmail_label_id: str
    gmail_label_name: str


class WatchedLabelCreate(BaseModel):
    """Activar una etiqueta para vigilar."""
    gmail_label_id: str
    gmail_label_name: str


class WatchedLabelOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    gmail_label_id: str
    gmail_label_name: str
    is_active: bool
    created_at: datetime

File: frontend/src/types/label.ts
export interface GmailLabelAvailable {
  gmail_label_id: string;
  gmail_label_name: string;
}

export interface WatchedLabel {
  id: number;
  gmail_label_id: string;
  gmail_label_name: string;
  is_active: boolean;
  created_at: string;
}
